servicetalk-concurrent-api-internal/src/test/java/io/servicetalk/concurrent/api/publisher/AbstractPublishAndSubscribeOnTest.java

```diff
@@ -92,7 +94,19 @@ public void onComplete() {
                 });
         allDone.await();
 
-        verifyCapturedThreads(capturedThreads);
+        return verifyCapturedThreads(capturedThreads);
+    }
+
+    private static void recordThread(AtomicReferenceArray<Thread> threads, final int index) {
```

Comment:
Thanks for sharing new `Matcher`! Can the `recordThread` and `verifyCapturedThreads` utilities also be shared? You can have a public utility in `servicetalk-concurrent-api-internal` test source, not necessary to move them to `servicetalk-test-resources`.

Reply:
per your suggestion I have added a utility in the `concurrent.api.internal` package and use it for the three AbstractPublishAndSubscribe variations